Test Code

mercurrie opened the pull request:
Added Flowchart,
Added ASCII art
Added Comments to main.py

mercurrie committed:
Create Blackjack_Flowchart.pdf

added Blackjack flowchart to folder

mercurrie committed:
add art and main

included comments in main.py

mercurrie committed:
Merge branch 'Test' of https://github.com/mercurrie/Black-Jack- into Test

mercurrie committed:
art.py

added art.py

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,87 @@
+        return 21
+    elif count > 21:
+        for num in list:
+            if num == 11:
```
This is good.  I thought initially that you were always converting 11 to 1 until I saw this condition.

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,87 @@
+
+  if userScore == dealerScore:
+    return "Draw 🙃"
+  elif dealerScore == 0:
```
You're checking for 0 to determine if the dealer won?

mercurrie replied:
originally had 0 set as the value for Blackjack. I ended up just going with 21 for less confusion. This was an oversight on my part

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,87 @@
+    return "Draw 🙃"
+  elif dealerScore == 0:
+    return "Lose, opponent has Blackjack 😱"
+  elif userScore == 0:
```
Same as above.  Maybe I'm mis-interpreting things.

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,87 @@
+    return count
+
+# function compareScrore takes in user and dealer's scores, compares them, and returns a string
+def compareScore(userScore, dealerScore):
```
PEP-8 (look it up) conventions for Python variables and functions is all `snake_case`.  Please fix.

michaelhousman replied:
You also need to type-cast all input variables like so:

def filter_object_fields(object: dict, included: List[str] = [], excluded: List[str] = []) -> dict:

It makes a world of difference when you're trying to debug.

mercurrie replied:
I included the snake_case, and will continue with that format from now on.
As for the type-casting, I couldn't find any relevant resources or examples where it is used in functions like the example above. Hopefully you can explain this to me in more detail at a later time

michaelhousman replied:
@mercurrie -- type-casting means that you identify the type (e.g., str, list) of the inputs and outputs like so:

def filter_object_fields(object, included, excluded):

VS:

def filter_object_fields(object: dict, included: List[str] = [], excluded: List[str] = []) -> dict:

What it does is allow the IDE to easily identify if there's any errors that might appear caused by incorrect type's before you run the code.  Use the function above as an example and see if you can specify what every input and output of every function should be.

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,87 @@
+    print(compareScore(userScore, dealerScore))
+    
+# call Blackjack function and clear console as long as User wants to continue playing
+while input("Would you like to play a game of Blackjack? Input 'y' for yes 'n' for no: ") == 'y':
```
You need to call your script with something like this at the bottom:

if __name__ == '__main__':
    test_list = [i for i in range(0, 100)]
    print([i for i in shard_list(test_list, 5)])

mercurrie replied:
Great feedback, included in code

michaelhousman reviewed (changes requested):
1. Really good start.  I'm impressed.
2. Like how you used functions here. I saw that you didn't in some of the other repos and you should fix that
3. Added some comments that I would fix

mercurrie committed:
update main.py

Change variables and functions to snake_case.
include: if __name__ == '__main__'.

0 was orinally meant for 'blackjack", changed into 21

michaelhousman commented:
I know you merged already but I do want to see you fix that one issue that was outstanding from the last review.  Can you take out a PR with those changes?
